asset/burn.go

```diff
@@ -0,0 +1,42 @@
+package asset
+
+import (
+	"bytes"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// DeriveBurnKey derives a provably un-spendable but unique key by tweaking the
+// public NUMS key with a tap tweak:
+//
+//	burnTweak = h_tapTweak(NUMSKey || outPoint || assetID || scriptKey)
+//	burnKey = NUMSKey + burnTweak*G
+//
+// The firstPrevID must be the PrevID from the first input that is being spent
+// by the virtual transaction that contains the burn.
+func DeriveBurnKey(firstPrevID PrevID) *btcec.PublicKey {
```

Comment:
Compared to the latest sketch, this doesn't consider if an asset is grouped or not. IIUC that's alright as outpoint + ID + scriptkey will be globally unique anyways, and we don't need this burn key tweak to map directly to an asset's location in the Tap tree.

Reply:
Hmm true, but yeah don't think it matters since the asset ID is unique.